java-api-core/src/com/kinvey/java/store/BaseDataStore.java

```diff
@@ -531,7 +532,18 @@ public KinveyPullResponse pullBlocking(Query query, int pageSize) throws IOExcep
         List<QueryCacheItem> queryCacheItems = queryCache.get(client.query().equals(Constants.QUERY, query.getQueryFilterMap().toString()));
         if (queryCacheItems.size() == 1) { //one is correct number of query cache item count for any request.
             QueryCacheItem cacheItem = queryCacheItems.get(0);
-            KinveyQueryCacheResponse<T> queryCacheResponse = networkManager.queryCacheGetBlocking(query, cacheItem.getLastRequestTime()).execute();
+            KinveyQueryCacheResponse<T> queryCacheResponse;
+            try {
+                queryCacheResponse = networkManager.queryCacheGetBlocking(query, cacheItem.getLastRequestTime()).execute();
+            } catch (KinveyJsonResponseException responseException) {
+                if (responseException.getDetails().getError().equals("MissingConfiguration")
+                        && responseException.getDetails().getDescription().equals("This feature is not properly configured for this app backend. Please configure it through the console first, or contact support for more information.")) {
```

Comment:
Would be better here to compare against status code instead of error description, since a change in the text of the error description would cause this to fail.

Reply:
Updated.